### PR TITLE
test: check pkglock is updated correctly

### PR DESCRIPTION
**Pull request by feelepxyz**
@larsgw awesome work on this fix! I've been trying to understand what it's doing and added a few tests to check the lockfile was updated correctly.

Let me know if this makes sense and is helpful!

Also fyi, pushed a fix for the failing spec in the `release-next` branch: https://github.com/npm/cli/pull/88

### CODE REVIEW

File: test/tap/install-dep-classification.js [hunk 1]
```diff
@@ -89,6 +105,47 @@ test('development dependency identification', function (t) {
       t.is(code, 0, 'no error code')
       t.is(stderr, '', 'no error output')
       t.notOk(fs.existsSync(path.join(devDir, 'node_modules')), 'did not install anything')
+      t.similar(JSON.parse(fs.readFileSync(path.join(optionalDir, 'package-lock.json'), 'utf8')), {
```
Comment (larsgw) [inline test/tap/install-dep-classification.js]: This should be `devDir` right?

File: test/tap/install-dep-classification.js [hunk 2]
```diff
@@ -89,6 +105,47 @@ test('development dependency identification', function (t) {
...
+        dependencies: {
+          a: {
+            version: 'file:../a-1.0.0.tgz',
+            optional: true,
```
Comment (larsgw) [inline test/tap/install-dep-classification.js]: And these ones changed accordingly to `dev: true`.

File: test/tap/install-dep-classification.js [hunk 3]
```diff
@@ -89,6 +105,47 @@ test('development dependency identification', function (t) {
...
+})
+
+test('default dependency identification', function (t) {
+  common.npm(
+    ['install'],
+    {cwd: optionalDir},
+    function (er, code, stdout, stderr) {
+      t.is(code, 0, 'no error code')
+      t.is(stderr, '', 'no error output')
+      t.similar(JSON.parse(fs.readFileSync(path.join(optionalDir, 'package-lock.json'), 'utf8')), {
```
Comment (larsgw) [inline test/tap/install-dep-classification.js:136]: This package lock also has `optional: true` on each dependency, did you omit those on purpose?

File: test/tap/install-dep-classification.js [hunk 4]
```diff
@@ -89,6 +105,47 @@ test('development dependency identification', function (t) {
...
+          b: {
+            version: 'file:../b-1.0.0.tgz',
+            optional: true
+          },
+          example: {
+            version: "1.0.0",
+            optional: true,
+          }
+        }
+      }, 'locks dependencies as optional')
```
Comment (larsgw) [inline test/tap/install-dep-classification.js]: And then this should be `dev` too.

**Review by larsgw — CHANGES_REQUESTED** [review]
Thanks for the tests, that side of the behaviour is definitely important, especially now that that's the biggest problem it solves. Here's some feedback (first time using the review tool!).

**Comment by feelepxyz** [comment]
@larsgw nice one, thanks for the feedback! Pushed the fixes 👌

**Review by larsgw — APPROVED** [review]

**Pull request merged**